examples/examples-cpu/snowflake/snowflake-dask.ipynb

```diff
@@ -0,0 +1,388 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Snowflake + Dask\n",
+    "\n",
+    "How to load data from a Snowflake table or query into a Dask dataframe"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Connect to Snowflake\n",
+    "\n",
+    "See [README](README.md) for more details on how to set up the credentials file."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import yaml\n",
+    "import snowflake.connector\n",
+    "\n",
+    "creds = yaml.full_load(open('/home/jovyan/snowflake_creds.yml'))\n",
+    "\n",
+    "# get connection info\n",
+    "conn_info = {\n",
+    "    'warehouse': 'COMPUTE_WH',\n",
+    "    'database': 'NYC_TAXI',\n",
+    "    'schema': 'PUBLIC',\n",
+    "    **creds,\n",
+    "}\n",
+    "conn = snowflake.connector.connect(**conn_info)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Setup query\n",
+    "\n",
+    "We need to set up a query that will return chunks of the full result based on a column in the data. These become our partitions in a Dask dataframe. We use a [binding for the Snowflake query](https://docs.snowflake.com/en/user-guide/python-connector-example.html#binding-data) so that we can pass different values at execution time."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "query = \"\"\"\n",
+    "SELECT *\n",
+    "FROM taxi_yellow\n",
+    "WHERE\n",
+    "    date(pickup_datetime) = %s\n",
+    "\"\"\""
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Validate the query is good with pandas"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "cur = conn.cursor().execute(query, '2019-01-01')\n",
+    "df = cur.fetch_pandas_all()\n",
+    "len(df), df.memory_usage().sum() / 1e6  # memory size in MB"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Initialize Dask cluster"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from dask.distributed import Client, wait\n",
+    "from dask_saturn import SaturnCluster\n",
+    "import time\n",
```

Comment:
do you have a strong preference for sprinkling the imports throughout the notebook, instead of collecting them at the top?

I like having all the imports at the top, because it means that you find out right away if you have any import issues. Leaving them through the code like this means you could get a few minutes into the notebook and then realize you have to go install something, which means you might have to start over.

I'd prefer to keep them all at the top, but if you disagree it's not worth blocking the PR over.

Reply:
At least for now I'd like to keep it as-is. I do to have them spread throughout the examples because they provide context for the different classes/functions that are used. The import issues are slightly mitigated because these are paired with images already that should have anything. Also, the sprinkling is how its done in most of the other examples so I would rather at least keep the same pattern for now, and if we want to change for future we can go through and make everything consistent.

Reply:
ok